systemservices/marketplace/mesg.yml

```diff
@@ -0,0 +1,414 @@
+            description: "The value in Wei."
+            type: String
+          gas:
+            name: "Gas"
+            description: "The maximum gas to use to execute this transaction"
+            type: Number
+          gasPrice:
+            name: "Gas price"
+            description: "The gas price in Wei."
+            type: String
+          data:
+            name: "Data"
+            description: "The data of the transaction."
+            type: String
+  createServiceOffer:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      price:
+        name: ""
+        description: ""
+        type: String
+      duration:
+        name: ""
+        description: ""
+        type: String
+      from: *inputFrom
+      gas: *inputGas
+      gasPrice: *inputGasPrice
+    outputs:
+      error: *outputError
+      success: *outputTransaction
+  createServiceVersion:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      hash:
+        name: ""
+        description: ""
+        type: String
+      manifest:
+        name: ""
+        description: ""
+        type: String
+      manifestProtocol:
+        name: ""
+        description: ""
+        type: String
+      from: *inputFrom
+      gas: *inputGas
+      gasPrice: *inputGasPrice
+    outputs:
+      error: *outputError
+      success: *outputTransaction
+  disableServiceOffer:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      offerIndex:
+        name: ""
+        description: ""
+        type: String
+      from: *inputFrom
+      gas: *inputGas
+      gasPrice: *inputGasPrice
+    outputs:
+      error: *outputError
+      success: *outputTransaction
+  purchase:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      offerIndex:
+        name: ""
+        description: ""
+        type: String
+      from: *inputFrom
+      gas: *inputGas
+      gasPrice: *inputGasPrice
+    outputs:
+      error: *outputError
+      success: *outputTransaction
+  transferServiceOwnership:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      newOwner:
+        name: ""
+        description: ""
+        type: String
+      from: *inputFrom
+      gas: *inputGas
+      gasPrice: *inputGasPrice
+    outputs:
+      error: *outputError
+      success: *outputTransaction
+  sendSignedTransaction:
+    name: ""
+    description: ""
+    inputs:
+      signedTransaction:
+        name: ""
+        description: ""
+        type: String
+    outputs:
+      error: *outputError
+      success:
+        data:
+          receipt:
+            type: Object
+  isAuthorized:
+    name: ""
+    description: ""
+    inputs:
+      sid: *inputSid
+      from: *inputFrom
+    outputs:
+      error: *outputError
+      success:
+        name: ""
+        description: ""
+        data:
+          authorized:
+            name: ""
+            description: ""
+            type: Boolean
+  checkForDeployment:
```

Comment:
what's the purpose of this one ?

Reply:
This task is called by the CLI or Core when a service need to be deployed.
It check if one of the account of the wallet has is authorize (has paid) the service.

Reply:
so it needs to be renamed. If this function is checking the authorization to access a service maybe `checkauthorisation` here the name sounds like we check if we can deploy and we are on the service marketplace so there is no notion of deployment on the core just deployment on the marketplace

Reply:
```suggestion
  checkAuthorization:
```